pymedphys/tests/dicom/structure/test_masking.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (C) 2020 Cancer Care Associates
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+import numpy as np
+import shapely.geometry
+
+from pymedphys._dicom.coords import xyz_axes_from_dataset
+from pymedphys._dicom.create import dicom_dataset_from_dict
+from pymedphys._dicom.dose import get_dose_grid_structure_mask
+
+
+@pytest.mark.pydicom
+def test_structure_dose_mask():
+    dx = 0.1
+    dy = 0.2
+
+    x0 = -3
+    y0 = -1
+
+    epsilon = 0.001
+    contour_name = "rectangle"
+
+    x_grid = np.arange(x0, 2, dx)
+    y_grid = np.arange(y0, 4, dy)
+
+    xx, yy = np.meshgrid(x_grid, y_grid)
+
+    contour_x = [0, 0, 1, 1]
+    contour_y = [0, 2, 2, 0]
+    contour_z = [0, 1, 2]
+
+    contour = np.vstack([contour_x, contour_y]).T
+
+    eroded_mask = shapely_based_masking_with_epsilon_buffer(xx, yy, contour, -epsilon)
+    buffered_mask = shapely_based_masking_with_epsilon_buffer(xx, yy, contour, epsilon)
+
+    structure_dataset, dose_dataset = convert_contours_to_dummy_dicom_files(
+        x_grid, y_grid, contour_x, contour_y, contour_z, contour_name
+    )
+
+    x_dicom, y_dicom, z_dicom = xyz_axes_from_dataset(dose_dataset)
+
+    assert np.allclose(x_dicom, x_grid)
+    assert np.allclose(y_dicom, y_grid)
+    assert np.allclose(z_dicom, contour_z)
+
+    dicom_mask = get_dose_grid_structure_mask(
+        contour_name, structure_dataset, dose_dataset
+    )
+
+    for i, _ in enumerate(contour_z):
+        # All points within the eroded mask are also within the dicom_mask
+        assert np.all(np.logical_and(eroded_mask, dicom_mask[i, :, :]) == eroded_mask)
+
+        # All points outside the buffered mask are also outside the dicom_mask
+        assert np.all(
+            np.logical_and(
+                np.logical_not(buffered_mask), np.logical_not(dicom_mask[i, :, :])
+            )
+            == np.logical_not(buffered_mask)
+        )
+
+
+def get_grid_spacing(array):
+    dx = np.unique(np.round(np.diff(array), 4))
+    assert len(dx) == 1
+
+    return dx
+
+
+def convert_contours_to_dummy_dicom_files(
```

Comment:
if it's intended to be public, there is enough going on here to include a docstring